android/app/src/main/java/org/cryptimeleon/incentive/app/domain/usecase/GetPromotionStatesUseCase.kt

```diff
@@ -0,0 +1,59 @@
+package org.cryptimeleon.incentive.app.domain.usecase
+
+import androidx.compose.runtime.rememberUpdatedState
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import org.cryptimeleon.incentive.app.domain.IBasketRepository
+import org.cryptimeleon.incentive.app.domain.ICryptoRepository
+import org.cryptimeleon.incentive.app.domain.IPromotionRepository
+import org.cryptimeleon.incentive.app.domain.model.PromotionState
+import org.cryptimeleon.incentive.app.domain.model.UpdateChoice
+import org.cryptimeleon.incentive.app.util.toBigIntVector
+import org.cryptimeleon.incentive.crypto.model.Token
+
+class GetPromotionStatesUseCase(
```

Comment:
what exactly is this class good for?
comment might be good

Reply:
Okay